util/pace/util/grid/generation.py

```diff
@@ -1529,6 +1560,64 @@ def rdyc(self) -> util.Quantity:
             gt4py_backend=self.dyc.gt4py_backend,
         )
 
+    def _init_cartesian(self):
+
+        domain_rad = PI / 16.0
+        lat_rad = self._deglat * PI / 180.0
+        lon_rad = 0.0
+
+        self._dx, self._dy = self._compute_dxdy_cartesian()
+        self._dx_agrid, self._dy_agrid = self._compute_dxdy_agrid_cartesian()
+        self._dx_center, self._dy_center = self._compute_dxdy_center_cartesian()
+        self._area = self._compute_area_cartesian()
+        self._area_c = self._compute_area_c_cartesian()
+
+        self._grid_64.data[:, :, :] = self._np.nan
+        slice_x, slice_y = self._tile_partitioner.subtile_slice(
+            self._rank, self._grid_64.dims, (self._npx, self._npy)
+        )
+
+        isd = slice_x.start - self._halo
+        ied = slice_x.stop + self._halo
+        isg = max(isd, 0)
+        ieg = min(ied, self._npx)
+        isl = isg - isd
+        iel = isl + ieg - isg
+
+        jsd = slice_y.start - self._halo
+        jed = slice_y.stop + self._halo
+        jsg = max(jsd, 0)
+        jeg = min(jed, self._npy)
+        jsl = jsg - jsd
+        jel = jsl + jeg - jsg
+
+        lon_frac = np.array(range(isg, ieg)) / (self._npx - 1) - 0.5
+        lon_frac = lon_frac[:, np.newaxis]
+        lat_frac = np.array(range(jsg, jeg)) / (self._npy - 1) - 0.5
+        lat_frac = lat_frac[np.newaxis, :]
+
+        self._grid_64.data[isl:iel, jsl:jel, 0] = lon_rad + lon_frac * domain_rad
+        self._grid_64.data[isl:iel, jsl:jel, 1] = lat_rad + lat_frac * domain_rad
+
+        self._agrid_64.data[:, :, 0] = lon_rad
+        self._agrid_64.data[:, :, 1] = lat_rad
+
+        self._init_cell_trigonometry_cartesian()
+
+        self._ec1, self._ec2 = self._calculate_center_vectors_cartesian()
+        self._ew1, self._ew2 = self._calculate_vectors_west_cartesian()
+        self._es1, self._es2 = self._calculate_vectors_south_cartesian()
+
+        # TODO: following lines just fill fields with nan
+        # presumably these aren't needed other than for testing
+        # so best to get rid of them to reduce memory pressure?
```

Comment:
+1

Can you pass a "fill_all_fields" option to the `__init__` default to `False` that's flipped for test?